More operator docs (#2107)
kroxylicious/kroxylicious: docs/modules/configuring/con-virtualkafkacluster-tcp.adoc

// file included in the following:
//
// kroxylicious-operator/assemblies/assembly-operator-deploy-a-proxy.adoc

[id='con-configuring-virtualkafkacluster-{context}']

> = A virtual cluster for on-cluster access without TLS

Reviewer: Wondering where the right place to add a note describing our limitation around only allowing a single port identifies broker ingress within a KafkaProxy is. This module is sort-of in the right area, but I guess it's more a part of the ingress concepts where we might enumerate the different ingress types.

> = A virtual cluster for on-cluster access without TLS

Reviewer 2: ```suggestion
= Virtual cluster configuration for in-cluster access without TLS
```

A `VirtualKafkaCluster` resource defines a logical Kafka cluster that is accessible to clients over the network.

The virtual cluster references the following:

* A `KafkaProxy` resource that the proxy is associated with.
* One or more `KafkaProxyIngress` resources that expose the virtual cluster to Kafka clients.
* A `KafkaService` resource that defined the backend Kafka cluster.
* Zero or more `KafkaProtocolFilter` resources that apply filters to the Kafka protocol traffic passing between clients and the backend Kafka cluster.

This example shows a `VirtualKafkaCluster`, exposing it to Kafka clients running on the same Kubernetes cluster.
It uses plain TCP (as opposed to TLS) as the transport protocol.

> .Example `VirtualKafkaCluster` configuration.

Reviewer 2: ```suggestion
.Example `VirtualKafkaCluster` configuration
```

[source,yaml]
----
kind: VirtualKafkaCluster
apiVersion: kroxylicious.io/v1alpha1
metadata:
  name: my-cluster
  namespace: my-proxy
spec:
  proxyRef: # <1>
    name: simple
  targetKafkaServiceRef: # <2>
    name: my-cluster
  ingresses:
    - ingressRef: # <3>
        name: cluster-ip
----

> <1> The `proxyRef` names the `KafkaProxy` resource that this virtual cluster is part of.

Reviewer 2: ```suggestion
<1> The `proxyRef` names the `KafkaProxy` associated with this virtual cluster.
```

Author: I think we should use whatever verb is the outcome of the [discussion above](https://github.com/kroxylicious/kroxylicious/pull/2107#discussion_r2072723428)

Reviewer 2: Thanks for addressing teh comments @tombentley 
Host sounds good to me.
I've made a suggestion about the glossary items.

Author: ```suggestion
<1> The `proxyRef` names the `KafkaProxy` hosting with this virtual cluster.
```

  It must be in the same namespace as the `VirtualKafkaCluster`.

> <2> The virtual cluster names the `KafkaService` to be proxied.

Reviewer 2: ```suggestion
<2> The `KafkaService` that proxies the virtual cluster.
```

Author: ```suggestion
<2> The `KafkaService` that is proxied by the virtual cluster.
```

  It must be in the same namespace as the `VirtualKafkaCluster`.

> <3> The virtual cluster can be exposed by one or more ingresses.

Reviewer 2: ```suggestion
<3> Ingresses to expose the virtual cluster.
```

  Each ingress names a `KafkaProxyIngress` which must be in the same namespace as the `VirtualKafkaCluster`.

> // Let's look at what the referenced `KafkaProxyIngress` would look like.

Reviewer 2: skipped this
